src/components/PopupConfirm.js

```diff
@@ -0,0 +1,26 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirm extends Popup {
+  constructor(popupSelector, handleButtonClick) {
+    super(popupSelector);
+    this._handleButtonClick = handleButtonClick;
+    this._button = this._popupEl.querySelector(".modal__button");
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._button.addEventListener("click", () => {
+      this._handleButtonClick(this._calledEl);
+    });
+  }
+
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```